### ISSUE TITLE
 Intel® Core™ Ultra processors NPU  Support 

### ISSUE THREAD

**Issue by mofanke** [ggml-org/llama.cpp#5079]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Feature Description

 Intel® Core™ Ultra processors now has released  , how can llama.cpp use that npu to fast up 

# Motivation

 Intel® Core™ Ultra processors deliver three dedicated engines (CPU, GPU, and NPU) to help unlock the power of AI
https://www.intel.com/content/www/us/en/products/docs/processors/core-ultra/core-ultra-series-1-product-brief.html

# Possible Implementation

OpenVINO™, WindowsML, DirectML, ONNX RT

**Comment by LazyGeniusMan** [comment]
Intel open source its NPU implementation library. Looks like it's using OpenVINO

https://github.com/intel/intel-npu-acceleration-library
https://intel.github.io/intel-npu-acceleration-library/cpp_reference.html

**Comment by paaspaas00** [comment]
Hi, any plans for this?

**Comment by asychev** [comment]
Does the fact this issue closed means there is no interest/plan for the future?

**Comment by sebastienbo** [comment]
Yeah I don't understand, running an LLM on NPU's has so much potential

**Comment by hgftrdw45ud67is8o89** [comment]
this should be fixed to have manual issue closing lmao.

**Comment by peeeto111** [comment]
Yes please use this potential

**Comment by peeeto111** [comment]
Please add support for NPU.

**Comment by Muzixin** [comment]
need intel npu support

**Comment by yusiwen** [comment]
Any updates?

**Comment by sebastienbo** [comment]
Yes, I was answering the same, especially now that we start seeing Intel lunar lake laptops with 45tops npus

**Comment by makinikm** [comment]
Any plans to reopen this by public request? As in all the above including mine...

**Comment by mikehenken** [comment]
It would be ideal to prioritize support for this. If nothing else, then to further development and practicality of CPU/NPU chips.

**Comment by reneleonhardt** [comment]
I guess if a bot closed this feature request automatically a year ago, then no maintainer is interested in prioritizing this... 😞

**Comment by izelnakri** [comment]
@ggerganov @slaren @ngxson Given there is no clear documentation or implementation to support this feature, could you re-open this issue? I think this is probably the most important ticket to work on for `llama.cpp` given how many developers use Intel Meteor Lake+, despite their level in terms of skill.

I managed to get `llama.cpp` and `whisper.cpp` running on `NixOS`, it wasn't easy to get the `nix develop` shell working and required a small patch to remove a compilation errors. Now I realized both projects don't support my NPU, so I cannot use `llama.cpp` or `whisper.cpp` on my laptops sadly and will resort to `OpenVINO` runtime until `llama.cpp` supports `NPU`s.

Same story most likely applies to cloud instances. Most cloud providers charge extortion level prices for GPUs, new and cheaper AI instances will likely be Intel chips(its just cheaper cost/performance wise), so if you guys support this feature, I can completely mirror my production environment locally on my computer and use the same set of tools(`llama.cpp` and `whisper.cpp`) on my machines, maybe even on my VR headset & phone eventually when I can get access to devices in my `qemu` VMs and run NixOS on them, circumventing Android & NPKs.

In other words first-class Intel NPU support via OpenVINO is I think essential(not only on the encoder like you guys did in the `whisper.cpp`)

**Comment by ggerganov** [comment]
I think there are some efforts from Intel to bring OpenVINO support to `llama.cpp`/`ggml`.

This seems to be something relevant: https://github.com/intel/ipex-llm

You might want to ask on their forums for more information: https://github.com/intel/ipex-llm/issues/13234

**Comment by paaspaas00** [comment]
What's the suggested path here? Is Intel NPU support going to be added? Can It be taken from ipex-llm? Could the code be merged between the projects?

**Comment by rajasrijan** [comment]
I would also be interested in this.

**Comment by GabrielVander** [comment]
Please, any updates?

@ggerganov @slaren @ngxson shouldn't this issue be re-opened?

**Comment by nobody-forever** [comment]
reopen?
